src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -261,26 +261,25 @@ public List<LayoutBlock> buildBlocks(UserDTO userDTO, boolean trialAccountActiva
         return blocks;
     }
 
-    private LayoutBlock buildCollapsedBlock(UserDTO userDTO, ActionId actionId) {
+    private LayoutBlock buildCollapsedBlock(UserDTO userDTO, boolean trialAccountActivated, ActionId actionId) {
         StringBuilder sb = new StringBuilder();
-        sb.append(userDTO.getEmail() + "\n" + userDTO.getCompanyName() + " (" + userDTO.getLicenseType().getShortName() + (withTrialAccountNote(userDTO, actionId) ? ", *TRIAL*)" : userDTO.isActivated() ? ")" : ")\n*NOT ACTIVATED*: "));
-        if (!userDTO.isActivated() && !withTrialAccountNote(userDTO, actionId)) {
-            if (withRejectionNote(userDTO, actionId)) {
-                sb.append("Sent rejection email");
-            } else if (withRejectAlumniAddressNote(userDTO, actionId)) {
-                sb.append("Rejected user due to alumni email address");
-            } else if (withUseCaseClarificationNote(userDTO, actionId)) {
-                sb.append("Sent use case clarification");
-            } else if (withForProfitClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on for-profit affiliation");
-            } else if (withAcademicClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on noninstitutional email");
-            } else if (withDuplicateUserClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on multiple account request");
-            } else if (withRegistrationInfoClarificationNote(userDTO, actionId)) {
-                sb.append("Clarified with user on registration info");
-            } else if (withLicenseOptionsNote(userDTO, actionId)) {
-                sb.append("Sent license options email");
+        sb.append(userDTO.getEmail() + "\n" + userDTO.getCompanyName() + " (" + userDTO.getLicenseType().getShortName()
+            + ((withNote(DropdownEmailOption.GIVE_TRIAL_ACCESS, userDTO, actionId) && !(userDTO.isActivated() && !trialAccountActivated)) ? ", *TRIAL*)" :
+            (userDTO.isActivated() ? ")" : ")\n*NOT ACTIVATED*: ")));
+        if (!userDTO.isActivated() && !withNote(DropdownEmailOption.GIVE_TRIAL_ACCESS, userDTO, actionId)) {
+            List<DropdownEmailOption> sentMails = new ArrayList<>();
+            for (DropdownEmailOption mailOption : DropdownEmailOption.values()) {
+                if (withNote(mailOption, userDTO, actionId))
+                    sentMails.add(mailOption);
+            }
+            if (!sentMails.isEmpty()) {
+                sentMails = sentMails.stream().sorted((mo1, mo2) -> mo1.getCategory() == EmailCategory.DENY ? 1 : 0).collect(Collectors.toList());
```

Comment:
can we just do 
```suggestion
                sentMails = sentMails.stream().sorted((mo1, mo2) -> mo1.getCategory() == EmailCategory.DENY).collect(Collectors.toList());
```

Reply:
I believe `sorted()` takes an overridden `compare()` predicate to sort, which should return a number (negative means less than, zero means equal, positive means greater than). I wanted clarification and license emails to show up first and then rejection email last to have some sort of chronological order, so I made rejection email greater than all others. I don't think boolean `true` will automatically be converted to positive number.

I just realized that license options email should be after clarification emails as well, that would make more sense. Will add that to the changes.